src/routes/ui/mod.rs

```diff
@@ -64,150 +71,90 @@ pub fn create_status(
     Ok(Redirect::to("/"))
 }
 
-fn render_status(db_conn: &db::Connection, status: &Status, link: bool) -> Result<Markup, Error> {
-    let meta_line = html !{
-        span {
-            ("published: ")
-            time datetime=(status.created_at.to_rfc3339()) (status.humanized_age())
-        }
-    };
-
-    let rendered = html! {
-        div.status {
-            header {
-                @if link {
-                    a href=(status.get_uri(db_conn)?) (meta_line)
-                } @else {
-                    (meta_line)
-                }
-            }
-            div {
-                @if let Some(cw) = &status.content_warning {
-                    @let collapse_id = format!("collapsible-{}", status.id);
-
-                    span (cw)
-                    input.collapse--toggle id=(collapse_id) type="checkbox";
-                    label.collapse--lbl-toggle for=(collapse_id) tabindex="0" { "Toggle CW" }
-                    div.content.collapse--content (status.text)
-                } @else {
-                    div.content (status.text)
-                }
-            }
-        }
-    };
-
-    Ok(rendered)
-}
-
 #[get("/users/<username>/statuses/<status_id>", format = "text/html")]
-pub fn status_page(username: String, status_id: u64, db_conn: db::Connection) -> Perhaps<Page> {
+pub fn status_page(
+    username: String,
+    status_id: u64,
+    db_conn: db::Connection,
+) -> Perhaps<StatusTemplate<'static>> {
     let account = try_resopt!(Account::fetch_local_by_username(&db_conn, username));
     let status = try_resopt!(Status::by_account_and_id(
         &db_conn,
         account.id,
         status_id as i64
     ));
 
-    let rendered = Page::new()
-        .title(format!(
-            "@{user}: {id}",
-            user = account.username,
-            id = status.id
-        ))
-        .content(render_status(&db_conn, &status, false)?);
-
-    Ok(Some(rendered))
-}
-
-#[derive(FromForm, Debug)]
-pub struct UserPageParams {
-    max_id: Option<i64>,
+    Ok(Some(StatusTemplate {
+        status:  status,
+        link:    false,
```

Comment:
i would _strongly_ prefer to figure out a way of implementing the status rendering such that `link` doesn't end up in the root of the template variables. perhaps we could use an askama macro?

or maybe we could just get rid of `_status.html` entirely, since implementations on the status page, versus those on timeline views, versus those on user pages are likely to diverge significantly for styling reasons

Reply:
I’d be pro just removing the “partial” entirely, that seems like the simplest path to cleanup.